### PR TITLE
Add inline progress bar for project and (some) algorithm procedures

### PR DESCRIPTION
**Pull request by adamnsch**
Depends on https://github.com/neo-technology/graph-analytics/pull/5334 to be merged for tests to pass

### CODE REVIEW

File: graphdatascience/query_runner/query_runner.py [hunk 1]
```diff
@@ -11,6 +12,9 @@ class QueryRunner(ABC):
     def run_query(self, query: str, params: Dict[str, Any] = {}) -> DataFrame:
         pass
 
+    def run_query_with_logging(self, query: str, params: Dict[str, Any] = {}) -> DataFrame:
+        return self.run_query(query, params)
+
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply graphdatascience/query_runner/query_runner.py:17]: Very good point! Should override this in the arrow query runner

[the remark this answers is not in the document]
Comment (adamnsch) [reply graphdatascience/query_runner/query_runner.py:17]: done

File: graphdatascience/query_runner/neo4j_query_runner.py [hunk 2]
```diff
@@ -38,6 +44,58 @@ def run_query(self, query: str, params: Dict[str, Any] = {}) -> DataFrame:
...
+                # * has not started yet,
+                # * has already completed.
+                if f"No task with job id `{job_id}` was found" in str(e):
+                    continue
+                else:
+                    raise e
+
+            progress_percent = progress["progress"][0]
+            if not progress_percent == "n/a":
+                task_name = progress["taskName"][0].split("|--")[-1][1:]
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply graphdatascience/query_runner/neo4j_query_runner.py:88]: agree

**Comment by adamnsch** [comment]
> I think this is very cool. I only think that there is a bug with respect to the arrow query runner, that would prevent logging when arrow is enabled.

thanks for the review :) have addressed the issue with the arrow query runner now

**Review by DarthMax — APPROVED** [review]
Looks good!
Some further ideas, we could also show the progress bar for the import, based on the number of nodes/rels imported. They are reported back to the client periodically.

**Comment by adamnsch** [comment]
> Looks good! Some further ideas, we could also show the progress bar for the import, based on the number of nodes/rels imported. They are reported back to the client periodically.

You mean the arrow import? That would be great :) We'll need to add logging for the arrow stuff eventually too for sure

**Pull request merged**